src/sql/workbench/contrib/executionPlan/common/executionPlanInput.ts

```diff
@@ -5,23 +5,41 @@
 
 import * as path from 'vs/base/common/path';
 import { URI } from 'vs/base/common/uri';
+import { localize } from 'vs/nls';
 import { EditorInput } from 'vs/workbench/common/editor/editorInput';
 import { EditorModel } from 'vs/workbench/common/editor/editorModel';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
 import { ITextFileService } from 'vs/workbench/services/textfile/common/textfiles';
 
 export class ExecutionPlanInput extends EditorInput {
 
 	public static ID: string = 'workbench.editorinputs.executionplan';
 	public static SCHEMA: string = 'executionplan';
+	private readonly editorNamePrefix = localize('epCompare.executionPlanEditorName', 'ExecutionPlan');
+	private _editorName: string;
 
-	private _content?: string;
 	public _executionPlanFileViewUUID: string;
 
 	constructor(
-		private _uri: URI,
+		private _uri: URI | undefined,
+		private _content: string | undefined,
```

Comment:
NIT: Can you make content an object that takes both the string and the extension. It will look cleaner IMO.

Reply:
We already have this interface https://github.com/microsoft/azuredatastudio/blob/4096037167a7fea5ab3c4efcd6a041dfcfa687d4/src/sql/azdata.proposed.d.ts#L1416